be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -1050,7 +1055,7 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
                 fragment_ctx);
 
         size_t source_dop = fragment_ctx->pipelines().back()->source_operator_factory()->degree_of_parallelism();
-        size_t sink_dop = request.pipeline_sink_dop();
+        size_t sink_dop = _calc_sink_dop(ExecEnv::GetInstance(), request);
```

Comment:
May consider the function returns uint32_t. It is an implicit overflow/underflow if returns a negative int32_t and assigned to size_t.

Reply:
The implementation ensures it outputs a positive number.

Reply:
so why not `uint32_t _calc_sink_dop()` if it can only returns a positive number?

Reply:
ok.